tests/integration/test_workflow_changes_validation.py

```
@@ -0,0 +1,554 @@
+                assert 'with' in checkout_step, \
+                    "Checkout in pull_request_target should specify 'with' parameters"
+    
+    def test_workflows_have_appropriate_permissions(self):
+        """Verify workflows follow least privilege principle."""
+        workflow_files = {
+            ".github/workflows/pr-agent.yml": ['contents', 'issues', 'pull-requests'],
+            ".github/workflows/label.yml": ['pull-requests'],
+            ".github/workflows/greetings.yml": ['issues', 'pull-requests'],
+        }
+        
+        for workflow_file, expected_perms in workflow_files.items():
```

Ruff (0.14.7)

Loop control variable `expected_perms` not used within loop body

Rename unused `expected_perms` to `_expected_perms`

(B007)